### PR TITLE
Added getContainerclient and getServiceClient

### LINKED ISSUE #17152: [FEATURE REQ] Ability to traverse blob hierarchy upwards. BlobClient->BlobContainer
Propagating from .NET SDK for parity https://github.com/Azure/azure-sdk-for-net/issues/16359 .

### PR DESCRIPTION
**Pull request by rickle-msft**
Resolves #17152

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/samples/java/com/azure/storage/blob/BlobAsyncClientJavaDocCodeSnippets.java [hunk 1]
```diff
@@ -383,6 +383,16 @@ public void getContainerName() {
         // END: com.azure.storage.blob.specialized.BlobAsyncClientBase.getContainerName
     }
 
+    /**
+     * Generates a code sample for using {@link BlobAsyncClient#getContainerAsyncClient()}
+     */
+    public void getContainerClient() {
+        // BEGIN: com.azure.storage.blob.specialized.BlobClientBase.getContainerAsyncClient
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/samples/java/com/azure/storage/blob/BlobAsyncClientJavaDocCodeSnippets.java:390]:
```suggestion
        // BEGIN: com.azure.storage.blob.specialized.BlobAsyncClientBase.getContainerAsyncClient
```

**Review by alzimmermsft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java [hunk 2]
```diff
@@ -217,6 +218,27 @@ public String getAccountName() {
         return this.accountName;
     }
 
+    /**
+     * Get an async client pointing to the account.
+     *
+     * @return {@link BlobServiceAsyncClient}
+     */
+    public BlobServiceAsyncClient getServiceAsyncClient() {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java:226]: Should this get a code sample too?

**Review by gapra-msft — APPROVED** [review]
Should we make issues to add this to datalake, file share and queue?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java (hunk 2 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java:226]: We don't have code samples for other getters on this type, but I can add some if we want

**Comment by rickle-msft** [comment]
We didn't in .NET, which is why I didn't do it here, but we can ask if we should expand this to other packages and we want to keep parity there

**Pull request merged**